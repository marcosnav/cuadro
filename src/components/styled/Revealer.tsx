import styled from 'styled-components';

interface IProps {
  height?: number;
  show?: boolean;
  easeTiming?: string;
  direction?: 'UP'|'DOWN';
}

const getHeight = (height?: number) => {
  return height ? height + 'px' : 'auto';
};

const getTransform = (direction: string = 'UP') => {
  const yValue = direction === 'UP' ? '42px' : '-42px';
  return `translateY(${yValue})`;
};

const getTransition = (easeTiming: string = '0.8') => {
  return `opacity ${easeTiming}s ease, transform ${easeTiming}s ease`;
};

const Revealer = styled.div<IProps>`
  height: ${({ height, show }) => show ? getHeight(height) : '0'};
  opacity: ${({ show }) => show ? '1' : '0'};
  transform: ${({ direction, show }) => show ? 'translateY(0)' : getTransform(direction)};
  transition: ${({ easeTiming }) => getTransition(easeTiming)};
`;

Revealer.displayName = 'Revealer';

export default Revealer;
