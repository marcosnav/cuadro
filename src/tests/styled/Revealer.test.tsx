import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import Revealer from './../../components/styled/Revealer';

describe('<Revealer />', () => {
  const component = mount(<Revealer height={100} />);

  test('renders properly', () => {
    expect(component).toHaveStyleRule('height', '0');
    expect(component).toHaveStyleRule('opacity', '0');
    expect(component).toHaveStyleRule('transform', 'translateY(42px)');
    expect(component).toHaveStyleRule('transition', 'opacity 0.8s ease,transform 0.8s ease');
    expect(component).toMatchSnapshot();
  });

  test('show: true', () => {
    component.setProps({ show: true });
    expect(component).toHaveStyleRule('height', '100px');
    expect(component).toHaveStyleRule('opacity', '1');
    expect(component).toHaveStyleRule('transform', 'translateY(0)');
    expect(component).toMatchSnapshot();
  });

  test('easeTiming', () => {
    component.setProps({ easeTiming: '0.6' });
    expect(component).toHaveStyleRule('transition', 'opacity 0.6s ease,transform 0.6s ease');
    expect(component).toMatchSnapshot();
  });

  test('direction DOWN', () => {
    component.setProps({ show: false, direction: 'DOWN' });
    expect(component).toHaveStyleRule('transform', 'translateY(-42px)');
    expect(component).toMatchSnapshot();
  });
});
