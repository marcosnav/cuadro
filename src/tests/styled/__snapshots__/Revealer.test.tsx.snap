// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Revealer /> direction DOWN 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  -webkit-transform: translateY(-42px);
  -ms-transform: translateY(-42px);
  transform: translateY(-42px);
  -webkit-transition: opacity 0.6s ease,-webkit-transform 0.6s ease;
  -webkit-transition: opacity 0.6s ease,transform 0.6s ease;
  transition: opacity 0.6s ease,transform 0.6s ease;
}

<Revealer
  direction="DOWN"
  easeTiming="0.6"
  height={100}
  show={false}
>
  <div
    className="c0"
    direction="DOWN"
    height={100}
  />
</Revealer>
`;

exports[`<Revealer /> easeTiming 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: opacity 0.6s ease,-webkit-transform 0.6s ease;
  -webkit-transition: opacity 0.6s ease,transform 0.6s ease;
  transition: opacity 0.6s ease,transform 0.6s ease;
}

<Revealer
  easeTiming="0.6"
  height={100}
  show={true}
>
  <div
    className="c0"
    height={100}
  />
</Revealer>
`;

exports[`<Revealer /> renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
  -webkit-transform: translateY(42px);
  -ms-transform: translateY(42px);
  transform: translateY(42px);
  -webkit-transition: opacity 0.8s ease,-webkit-transform 0.8s ease;
  -webkit-transition: opacity 0.8s ease,transform 0.8s ease;
  transition: opacity 0.8s ease,transform 0.8s ease;
}

<Revealer
  height={100}
>
  <div
    className="c0"
    height={100}
  />
</Revealer>
`;

exports[`<Revealer /> show: true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: opacity 0.8s ease,-webkit-transform 0.8s ease;
  -webkit-transition: opacity 0.8s ease,transform 0.8s ease;
  transition: opacity 0.8s ease,transform 0.8s ease;
}

<Revealer
  height={100}
  show={true}
>
  <div
    className="c0"
    height={100}
  />
</Revealer>
`;
